CounterApp/App/AppCore.swift

```diff
@@ -0,0 +1,55 @@
+//
+//  AppCore.swift
+//  CounterApp
+//
+//  Created by Wenjuan Li on 2022/7/12.
+//
+
+import ComposableArchitecture
+import Foundation
+
+struct AppState: Equatable {
+    var counter: CounterState
+    var lock: LockState
+}
+
+enum AppAction: Equatable {
+    case counter(CounterAction)
+    case lock(LockAction)
+}
+
+struct AppEnviroment {}
+
+let appReducer = Reducer<AppState, AppAction, AppEnviroment>.combine(
+    counterReducer.pullback(
+        state: \AppState.counter,
+        action: /AppAction.counter,
+        environment: { _ in CounterEnviroment.defaultEnv() }
+    ),
+    lockReducer.pullback(
+        state: \AppState.lock,
+        action: /AppAction.lock,
+        environment: { _ in LockEnvironment.defaultEnv() }
+    ),
+    Reducer { _, action, _ in
+        switch action {
+        default:
+            return .none
+        }
+    }
```

Comment:
There's no need to define this `empty` reducer if we are not trying to perform any operation or return any side effect as a consequence of receiving either a `counter` or `lock` action. Also wanted to point out that the library comes with an `empty` reducer already. It is useful on the Previews or snapshot tests where some times you don't want to create a store with the real reducer.

Reply:
@nanocba Thanks! Fixed.